server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -160,7 +183,10 @@ void doRun() {
             // Stop cancelling tasks if there are no tokens in either of the two token buckets.
             if (rateLimitReached && ratioLimitReached) {
                 logger.debug("task cancellation limit reached");
-                state.incrementLimitReachedCount();
+                SearchBackpressureState searchBackpressureState = searchBackpressureStates.get(
```

Comment:
SearchBackpressureState searchBackpressureState = searchBackpressureStates.get(taskCancellation.getTask());
if (searchBackpressureState != null) {
    searchBackpressureState.incrementLimitReachedCount();
}

Reply:
Yep, this is much more readable.

Reply:
@reta What about the mocked tasks? When we run `mockedTask.getClass()`, it returns `org.opensearch.action.search.SearchShardTask$MockitoMock$2odFeVao`, and `searchBackpressureStates.get()` returns null as it doesn't find corresponding entry in the map.
Should we use `isAssignableFrom()` to handle this situation?

Reply:
@PritLadani why do you need to mock SearchShardTask and SearchTask? Those are quite simple classes that could be used as-is. But back to you question - yes, it seems like using `instanceOf` is safer:

```
searchBackpressureStates
            .entrySet()
            .stream()
            .filter(e -> e.getKey().isInstance(taskCancellation))
            .map(Map.Entry::getValue)
            .findAny();
```